### PR TITLE
add-repositories

### LINKED ISSUE #2: create-repo-layer
1. UserRepo.
2. MessRepo.
3. PostRepo.
4. FriendStatusRepo.

### PR DESCRIPTION
**Pull request by Iliaz41**
resolved #2

---

### Detailed summary
- Deleted `posgram/Models/Mess.cs` and `posgram/Repository/IRepository/IPostRepository.cs`.
- Added `using posgram.Models;` and `namespace posgram.Repository.IRepository` in `posgram/Repository/IRepository/IPostRepository.cs`.
- Added `using posgram.Models;` and `namespace posgram.Repository.IRepository` in `posgram/Repository/IRepository/IFriendStRepository.cs`.
- Added `using posgram.Models;`, `using System.Net;`, and `namespace posgram.Repository.IRepository` in `posgram/Repository/IRepository/IUserRepository.cs`.
- Removed `DbSet<Mess> Messes` from `posgram/Data/ApplicationDbContext.cs`.
- Added `using Microsoft.EntityFrameworkCore;`, `using posgram.Data;`, `using posgram.Models;`, and `using posgram.Repository.IRepository;` in `posgram/Repository/PostRepository.cs`.
- Added `using Microsoft.EntityFrameworkCore;`, `using posgram.Data;`, `using posgram.Models;`, and `using posgram.Repository.IRepository;` in `posgram/Repository/FriendStRepository.cs`.
- Added `using Microsoft.EntityFrameworkCore;`, `using posgram.Data;`, `using posgram.Models;`, and `using posgram.Repository.IRepository;` in `posgram/Repository/UserRepository.cs`.
- Implemented methods `GetAll()`, `GetById()`, `Create()`, `Update()`, and `Delete()` in `posgram/Repository/PostRepository.cs`.
- Implemented methods `GetAll()`, `GetById()`, `Create()`, `Update()`, and `Delete()` in `posgram/Repository/FriendStRepository.cs`.
- Implemented methods `GetAll()`, `GetById()`, `Create()`, `Update()`, `Delete()`, and `IsUserExsits()` in `posgram/Repository/UserRepository.cs`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: posgram/Repository/FriendStRepository.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        public async Task<FriendSt> GetById(long id)
+        {
+            FriendSt friendSt = await _dbContext.FriendSts.FindAsync(id);
+            return friendSt;
+        }
+
+        public async Task Save()
+        {
```
Comment (hizmailovich) [inline posgram/Repository/FriendStRepository.cs:40]: In my opinion, this method can be private, what do you think?

File: posgram/Repository/IRepository/IFriendStRepository.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+namespace posgram.Repository.IRepository
+{
+    public interface IFriendStRepository
+    {
+        Task<List<FriendSt>> GetAll();
+        Task<FriendSt> GetById(long id);
+        Task Create(FriendSt entity);
+        Task Update(FriendSt entity);
+        Task Delete(FriendSt entity);
+        Task Save();
```
Comment (hizmailovich) [inline posgram/Repository/IRepository/IFriendStRepository.cs:12]: Why do we need save method here?

Comment (Iliaz41) [reply posgram/Repository/IRepository/IFriendStRepository.cs:12]: All repositories have this method for saving changes in db. What do you mean? May be you mean remove this method from interface?

File: posgram/Repository/FriendStRepository.cs (hunk 1 above)
Comment (Iliaz41) [reply posgram/Repository/FriendStRepository.cs:40]: I have been thinking about this, may be.

File: posgram/Repository/IRepository/IFriendStRepository.cs (hunk 2 above)
Comment (hizmailovich) [reply posgram/Repository/IRepository/IFriendStRepository.cs:12]: yes, we don't use this method outside, do we?

Comment (Iliaz41) [reply posgram/Repository/IRepository/IFriendStRepository.cs:12]: Yep, no problem

File: posgram/Repository/FriendStRepository.cs (hunk 1 above)
Comment (hizmailovich) [reply posgram/Repository/FriendStRepository.cs:40]: let's make this method private or replace this call `await _dbContext.SaveChangesAsync();` to update and save methods

Comment (Iliaz41) [reply posgram/Repository/FriendStRepository.cs:40]: Okey.

**Review by hizmailovich — APPROVED** [review]

**Comment by Iliaz41** [comment]
@rultor merge

**Pull request merged**